File: Difficulty: Medium/Non-overlapping Intervals/nonoverlapping-intervals.java

class Solution {
    static int minRemoval(int intervals[][]) {
        Arrays.sort(intervals, (a,b) -> Integer.compare(a[1], b[1]));
        
        int res = 0;
        int prevEnd = intervals[0][1];
        
        for(int i=1; i<intervals.length; i++){
            if(prevEnd > intervals[i][0]){
                res++;
            }
            else{
                prevEnd = intervals[i][1];
            }
        }
        
        return res;
    }
}

File: Difficulty: Medium/Non-overlapping Intervals/nonoverlapping-intervals.py
#User function Template for python3

class Solution:
    def minRemoval(self, intervals):
        intervals.sort(key = lambda x : x[1])
        
        res = 0
        prevEnd = intervals[0][1]
        
        for i in range(1, len(intervals)):
            if prevEnd > intervals[i][0]:
                res += 1
            else:
                prevEnd = intervals[i][1]
                
        return res